trade-finance-manager-api/src/graphql/reducers/mappings/facilities/mapUkefExposure.js

```diff
@@ -1,17 +1,47 @@
+const { getUnixTime } = require('date-fns');
 const { formattedNumber } = require('../../../../utils/number');
+const api = require('../../../../v1/api');
+const { calculateNewFacilityValue, calculateUkefExposure, latestAmendmentValueAccepted } = require('../../../helpers/amendment.helpers');
 
-const mapUkefExposure = (facilityTfm) => {
-  const {
-    ukefExposure,
-    ukefExposureCalculationTimestamp,
-  } = facilityTfm;
+// maps ukef exposure on original value or latest amended value
+const mapUkefExposure = async (facilityTfm, facility) => {
+  if (facilityTfm) {
+    const {
+      ukefExposure,
+      ukefExposureCalculationTimestamp,
+      exchangeRate,
+    } = facilityTfm;
 
-  const formattedUkefExposure = formattedNumber(ukefExposure);
+    // sets facility exposure values and timestamp
+    let formattedUkefExposure = formattedNumber(ukefExposure);
+    let ukefExposureCalculationTimestampValue = ukefExposureCalculationTimestamp;
 
-  return {
-    exposure: `GBP ${formattedUkefExposure}`,
-    timestamp: `${ukefExposureCalculationTimestamp}`,
-  };
+    if (facility && facility?._id) {
+      const { _id } = facility;
+
+      const latestCompletedAmendment = await api.getLatestCompletedAmendment(_id);
+
+      if (latestCompletedAmendment?.amendmentId && latestCompletedAmendment?.value && latestAmendmentValueAccepted(latestCompletedAmendment)) {
+        const { coverPercentage, coveredPercentage } = facility.facilitySnapshot;
+
+        // BSS is coveredPercentage while GEF is coverPercentage
+        const coverPercentageValue = coverPercentage || coveredPercentage;
+
+        const valueInGBP = calculateNewFacilityValue(exchangeRate, latestCompletedAmendment);
+        const ukefExposureValue = calculateUkefExposure(valueInGBP, coverPercentageValue);
+
+        // sets new exposure value based on amendment value
+        formattedUkefExposure = formattedNumber(ukefExposureValue);
+        ukefExposureCalculationTimestampValue = getUnixTime(new Date());
+      }
+    }
+
+    return {
+      exposure: `GBP ${formattedUkefExposure}`,
```

Comment:
is this always in GBP? and can we use a constant for it?

Reply:
changed all